eventing/samples/k8s-events/README.md

```diff
@@ -0,0 +1,267 @@
+
+## Prerequisites
+
+- A Kubernetes cluster with Knative installed. Follow the
+  [installation instructions](https://github.com/knative/docs/blob/master/install/README.md)
+  if you need to create one.
+- [Docker](https://www.docker.com/) installed and running on your local machine,
+  and a Docker Hub account configured (you'll use it for a container registry).
+- The core Knative eventing tools installed. You can install them with:
+  ```shell
+  kubectl apply -f https://storage.googleapis.com/knative-releases/latest/release-eventing.yaml
+  ```
+
+## Configuring Knative
+
+To use this sample, you'll need to install the `stub` ClusterBus and the
+`k8sevents` EventSource.
+
+```shell
+kubectl apply -f https://storage.googleapis.com/knative-releases/latest/release-eventing-clusterbus-stub.yaml
+kubectl apply -f https://storage.googleapis.com/knative-releases/latest/release-eventing-source-k8sevents.yaml
+```
+
+## Granting permissions
+
+Because the `k8sevent` EventSource needs to create a Deployment and call the
+Kubernetes watch API on events, you'll need to provision a special
+ServiceAccount with the necessary permissions.
+
+The `serviceaccount.yaml` file provisions a service account, creates a role
+which can create Deployments in the `default` namespace and can view all
+Kubernetes resources. In a production environment, you might want to limit the
+access of this service account to only specific namespaces.
+
+```shell
+kubectl apply -f serviceaccount.yaml
+```
+
+## Build and deploy the sample
+
+1.  Use Docker to build the sample code into a container. To build and push with
+    Docker Hub, run these commands replacing {username} with your Docker Hub
+    username. Run the following from the _root_ of the `knative/docs` repo:
+
+    ```shell
+    # Build the container on your local machine
+    docker build -t {username}/k8s-events --file=eventing/samples/k8s-events/Dockerfile .
+
+    # Push the container to docker registry
+    docker push {username}/k8s-events
+    ```
+
+1.  After the build has completed and the container is pushed to Docker Hub, you
+    can deploy the function into your cluster. **Ensure that the container image
+    value in function.yaml matches the container you built in the previous step.**
+    Apply the configuration using `kubectl`:
+
+    ```shell
+    kubectl apply -f eventing/samples/k8s-events/function.yaml
+    ```
+
+1.  Check that your service is running using:
+
+    ```shell
+    kubectl get services.serving.knative.dev -o "custom-columns=NAME:.metadata.name,READY:.status.conditions[2].status,REASON:.status.conditions[2].message"
+    NAME              READY     REASON
+    read-k8s-events   True      <none>
+    ```
+
+1.  Create the flow sending Kubernetes Events to the service:
+
+    ```shell
+    kubectl apply -f eventing/samples/k8s-events/flow.yaml
+    ```
+
+1.  You can read the function logs using Kibana. To access Kibana, you need to
```

Comment:
Might be worth a footnote mentioning that Kibana isn't installed in the lite or no-mon installs and that`kubectl logs` is an alternative.